File: flutter_inappwebview_android/android/src/main/kotlin/dev/nosferatu500/inappwebview/in_app_browser/InAppBrowserDelegate.kt
package dev.nosferatu500.inappwebview.in_app_browser

import android.app.Activity

interface InAppBrowserDelegate {
  fun getActivity(): Activity?
  fun getActivityResultListeners(): MutableList<ActivityResultListener>
  fun didChangeTitle(title: String?)
  fun didStartNavigation(url: String?)
  fun didUpdateVisitedHistory(url: String?)
  fun didFinishNavigation(url: String?)
  fun didFailNavigation(url: String?, errorCode: Int, description: String?)
  fun didChangeProgress(progress: Int)
}

File: flutter_inappwebview_ios/ios/flutter_inappwebview_ios/Sources/flutter_inappwebview_ios/InAppBrowser/InAppBrowserDelegate.swift
//
//  InAppBrowserDelegate.swift
//  flutter_inappwebview
//
//  Created by Lorenzo Pichilli on 14/02/21.
//

import Foundation

/// `@MainActor` for the same reason as `Disposable`: the only conformer is
/// `InAppBrowserWebViewController`, a `UIViewController`, and every callback here is driven from a
/// `WKNavigationDelegate` method. Leaving the protocol nonisolated made the conformance illegal
/// under Swift 6 — *"crosses into main actor-isolated code and can cause data races"*.
@MainActor
public protocol InAppBrowserDelegate {
    func didChangeTitle(title: String?)
    func didStartNavigation(url: URL?)
    func didUpdateVisitedHistory(url: URL?)
    func didFinishNavigation(url: URL?)
    func didFailNavigation(url: URL?, error: Error)
    func didChangeProgress(progress: Double)
}
